### PR TITLE
Refactor gpArray to Clarify Usage

### PR DESCRIPTION
**Pull request by slari-pivotal**
The gpArray use of GpDB and Segment classes was confusing. This change renames GpDB to Segment and Segment to SegmentPair to clarify usage. Its a big diff, but a simple, repeating change.

We also refactored SegmentPair to conform to  our usage of only one mirror per primary.

### CODE REVIEW

File: gpMgmt/bin/gppylib/gparray.py [hunk 1]
```diff
@@ -8,9 +8,9 @@
     Contains three classes representing configuration information of a
     Greenplum array:
 
-      GpArray - The primary interface - collection of all GpDB within an array
-      GpDB    - represents configuration information for a single dbid
-      Segment - collection of all GpDB with the same content id
+      GpArray - The primary interface - collection of all Segment within an array
+      Segment    - represents configuration information for a single dbid
+      Segment - collection of all Segment with the same content id
```
Comment (Chibin) [inline gpMgmt/bin/gppylib/gparray.py]: Is this right?

File: gpMgmt/bin/gppylib/gparray.py [hunk 2]
```diff
@@ -1648,13 +1645,13 @@ def get_max_contentid(self, includeExpansionSegs=False):
...
-        return len(self.segments)
+        return len(self.segmentPairs)
 
     # --------------------------------------------------------------------
     def get_min_primary_port(self):
         """Returns the minimum primary segment db port"""
-        min_primary_port = self.segments[0].primaryDB.port
-        for seg in self.segments:
+        min_primary_port = self.segmentPairs[0].primaryDB.port
+        for seg in self.segmentPairs:
```
Comment (Chibin) [inline gpMgmt/bin/gppylib/gparray.py]: I see in some parts of the code where we use `for segPair in self.segmentPairs` and some that uses `for seg in self.segmentPairs`

**Comment by skahler-pivotal** [comment]
I'm a little bit concerned about the fact that I think we are using the verbiage of Segment for what is the Master and Standby Master. While I think it makes it much more clear for everything else in the cluster I want to make sure we aren't assigning a name that will make someone assume the role of a member that is not appropriate.

**Comment by slari-pivotal** [comment]
@skahler-pivotal we agree that SegmentPair is maybe too generic to use for a master/standby pair. But we're having a hard time thinking of a specific problem example. Do you have a case in mind? Should we add a chore for that refactor?

**Comment by slari-pivotal** [comment]
@chibin made the changes you suggested, thanks! There are still some places where we're using vars called 'seg' but those seemed appropriate to us, actually single segments.

**Comment by skahler-pivotal** [comment]
@slari-pivotal I think it's a good to run with. If it does become an issue we can address it. It is more clear than what we have now.

**Pull request merged**